File: services/auth/src/main/java/com/example/iter/auth/domain/repository/OAuthAccountRepository.java
package com.example.iter.auth.domain.repository;

import com.example.iter.auth.domain.entity.OAuthAccount;
import com.example.iter.auth.domain.entity.OAuthProvider;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface OAuthAccountRepository extends JpaRepository<OAuthAccount, Long> {

    Optional<OAuthAccount> findByProviderAndProviderUserId(
            OAuthProvider provider,
            String providerUserId
    );

    boolean existsByUserIdAndProvider(Long userId, OAuthProvider provider);
}

File: services/auth/src/main/kotlin/com/example/iter/auth/domain/repository/OAuthAccountRepository.kt
package com.example.iter.auth.domain.repository

import com.example.iter.auth.domain.entity.OAuthAccount
import com.example.iter.auth.domain.entity.OAuthProvider
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface OAuthAccountRepository : JpaRepository<OAuthAccount, Long> {

    fun findByProviderAndProviderUserId(provider: OAuthProvider, providerUserId: String): Optional<OAuthAccount>

    fun existsByUserIdAndProvider(userId: Long, provider: OAuthProvider): Boolean
}
